From: Thomas Renninger <trenn@suse.com>
Subject: Remove fcf-protection=full
References: 
Patch-Mainline: 
Git-commit: c06b22050797e630f41250ddee7a7f67eb1f3912
Git-repo: git@github.com:watologo1/intel-cmt-cat.git.git

This option causes grief with at least some gcc versions:

built failure ->"cc: error: unrecognized command line option '-fcf-
protection=full'; did you mean '-fstack-protector-all'?"

and some archs (i586):

 cc1: error: ‘-fcf-protection’ is not compatible with this target

According to our gcc maintainer (translated):
Whatever the reason is that they require gcc version 9..., maybe  -fcf-protection,
but simply throw it out...

If there is no urgent reason, better get rid of this one.

Signed-off-by: Thomas Renninger <trenn@suse.de>


Signed-off-by:  <trenn@suse.com>
diff --git a/examples/c/CAT_MBA/Makefile b/examples/c/CAT_MBA/Makefile
index 6b5ceb7..02adce8 100644
--- a/examples/c/CAT_MBA/Makefile
+++ b/examples/c/CAT_MBA/Makefile
@@ -63,8 +63,7 @@ IS_GCC = $(shell $(CC) -v 2>&1 | grep -c "^gcc version ")
 ifeq ($(IS_GCC),1)
 CFLAGS += -fno-strict-overflow \
     -fno-delete-null-pointer-checks \
-    -fwrapv \
-    -fcf-protection=full
+    -fwrapv
 endif
 
 # Build targets and dependencies
diff --git a/examples/c/CMT_MBM/Makefile b/examples/c/CMT_MBM/Makefile
index 650f37e..4090be9 100644
--- a/examples/c/CMT_MBM/Makefile
+++ b/examples/c/CMT_MBM/Makefile
@@ -63,8 +63,7 @@ IS_GCC = $(shell $(CC) -v 2>&1 | grep -c "^gcc version ")
 ifeq ($(IS_GCC),1)
 CFLAGS += -fno-strict-overflow \
     -fno-delete-null-pointer-checks \
-    -fwrapv \
-    -fcf-protection=full
+    -fwrapv
 endif
 
 # Build targets and dependencies
diff --git a/examples/c/PSEUDO_LOCK/Makefile b/examples/c/PSEUDO_LOCK/Makefile
index c0cdf66..37e7a60 100644
--- a/examples/c/PSEUDO_LOCK/Makefile
+++ b/examples/c/PSEUDO_LOCK/Makefile
@@ -63,8 +63,7 @@ IS_GCC = $(shell $(CC) -v 2>&1 | grep -c "^gcc version ")
 ifeq ($(IS_GCC),1)
 CFLAGS += -fno-strict-overflow \
     -fno-delete-null-pointer-checks \
-    -fwrapv \
-    -fcf-protection=full
+    -fwrapv
 endif
 
 # Build targets and dependencies
diff --git a/lib/Makefile b/lib/Makefile
index d171892..04eecc1 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -72,8 +72,7 @@ ifeq ($(IS_GCC),1)
 CFLAGS += -fno-strict-overflow \
     -fno-delete-null-pointer-checks \
     -fwrapv \
-    -fstack-protector-strong \
-    -fcf-protection=full
+    -fstack-protector-strong
 endif
 
 # so or static build
diff --git a/pqos/Makefile b/pqos/Makefile
index c1f1c0b..868e81a 100644
--- a/pqos/Makefile
+++ b/pqos/Makefile
@@ -45,8 +45,7 @@ CFLAGS = -I$(LIBDIR) \
 	-Wmissing-declarations -Wold-style-definition -Wpointer-arith \
 	-Wcast-qual -Wundef -Wwrite-strings \
 	-Wformat -Wformat-security -fstack-protector -fPIE \
-	-Wunreachable-code -Wsign-compare -Wno-endif-labels \
-	-fcf-protection=full
+	-Wunreachable-code -Wsign-compare -Wno-endif-labels
 ifneq ($(EXTRA_CFLAGS),)
 CFLAGS += $(EXTRA_CFLAGS)
 endif
@@ -70,8 +69,7 @@ ifeq ($(IS_GCC),1)
 CFLAGS += -fno-strict-overflow \
     -fno-delete-null-pointer-checks \
     -fwrapv \
-    -fstack-protector-strong \
-    -fcf-protection=full
+    -fstack-protector-strong
 endif
 
 ifeq ($(shell uname), FreeBSD)
diff --git a/rdtset/Makefile b/rdtset/Makefile
index 7b34594..36f3343 100644
--- a/rdtset/Makefile
+++ b/rdtset/Makefile
@@ -45,8 +45,7 @@ CFLAGS = -I$(LIBDIR) \
 	-Wcast-qual -Wundef -Wwrite-strings \
 	-Wformat -Wformat-security -fstack-protector -fPIE \
 	-Wunreachable-code -Wsign-compare -Wno-endif-labels \
-	-D_GNU_SOURCE \
-	-fcf-protection=full
+	-D_GNU_SOURCE
 ifneq ($(EXTRA_CFLAGS),)
 CFLAGS += $(EXTRA_CFLAGS)
 endif
@@ -70,8 +69,7 @@ ifeq ($(IS_GCC),1)
 CFLAGS += -fno-strict-overflow \
 	-fno-delete-null-pointer-checks \
 	-fwrapv \
-	-fstack-protector-strong \
-	-fcf-protection=full
+	-fstack-protector-strong
 endif
 
 # DEBUG build
diff --git a/tools/membw/Makefile b/tools/membw/Makefile
index fbe25ab..4246af9 100644
--- a/tools/membw/Makefile
+++ b/tools/membw/Makefile
@@ -49,8 +49,7 @@ CFLAGS=-W -Wall -Wextra -Wstrict-prototypes -Wmissing-prototypes \
 	-Wcast-qual -Wundef -Wwrite-strings \
 	-Wformat -Wformat-security -fstack-protector -fPIE \
 	-Wunreachable-code -Wsign-compare -Wno-endif-labels \
-	-Winline -msse4.2 \
-	-fcf-protection=full
+	-Winline -msse4.2
 
 ifeq ($(DEBUG),y)
 CFLAGS += -O0 -g -DDEBUG
@@ -77,8 +76,7 @@ CFLAGS += -fno-strict-overflow \
     -fno-delete-null-pointer-checks \
     -fwrapv \
     -fno-expensive-optimizations \
-    -fstack-protector-strong \
-    -fcf-protection=full
+    -fstack-protector-strong
 endif
 
 SRCS = $(sort $(wildcard *.c))
diff --git a/unit-test/lib/Makefile b/unit-test/lib/Makefile
index 3bf18a7..b472489 100644
--- a/unit-test/lib/Makefile
+++ b/unit-test/lib/Makefile
@@ -34,8 +34,7 @@ ifeq ($(IS_GCC),1)
 CFLAGS += -fno-strict-overflow \
 	-fno-delete-null-pointer-checks \
 	-fwrapv \
-	-fstack-protector-strong \
-	-fcf-protection=full
+	-fstack-protector-strong
 endif
 
 # common function wrap
diff --git a/unit-test/mock/Makefile b/unit-test/mock/Makefile
index d2ee6e0..aad7603 100644
--- a/unit-test/mock/Makefile
+++ b/unit-test/mock/Makefile
@@ -68,8 +68,7 @@ IS_GCC = $(shell $(CC) -v 2>&1 | grep -c "^gcc version ")
 ifeq ($(IS_GCC),1)
 CFLAGS += -fno-strict-overflow \
     -fno-delete-null-pointer-checks \
-    -fwrapv \
-    -fcf-protection=full
+    -fwrapv
 endif
 
 # so or static build
diff --git a/unit-test/output/Makefile b/unit-test/output/Makefile
index 72f7ed5..d4ae752 100644
--- a/unit-test/output/Makefile
+++ b/unit-test/output/Makefile
@@ -65,8 +65,7 @@ IS_GCC = $(shell $(CC) -v 2>&1 | grep -c "^gcc version ")
 ifeq ($(IS_GCC),1)
 CFLAGS += -fno-strict-overflow \
     -fno-delete-null-pointer-checks \
-    -fwrapv \
-    -fcf-protection=full
+    -fwrapv
 endif
 
 # so or static build
diff --git a/unit-test/pqos/Makefile b/unit-test/pqos/Makefile
index a980c19..509ee31 100644
--- a/unit-test/pqos/Makefile
+++ b/unit-test/pqos/Makefile
@@ -69,8 +69,7 @@ ifeq ($(IS_GCC),1)
 CFLAGS += -fno-strict-overflow \
     -fno-delete-null-pointer-checks \
     -fwrapv \
-    -fstack-protector-strong \
-    -fcf-protection=full
+    -fstack-protector-strong
 endif
 
 CFLAGS += -g -ggdb -O0
